consensus/bor/bor.go

```diff
@@ -892,13 +895,16 @@ func (c *Bor) GetCurrentSpan(headerHash common.Hash) (*Span, error) {
 }
 
 // GetCurrentValidators get current validators
-func (c *Bor) GetCurrentValidators(snapshotNumber uint64, blockNumber uint64) ([]*Validator, error) {
+func (c *Bor) GetCurrentValidators(headerHash common.Hash, blockNumber uint64) ([]*Validator, error) {
 	// block
-	blockNr := rpc.BlockNumber(snapshotNumber)
+	blockNr := rpc.BlockNumberOrHashWithHash(headerHash, false)
 
 	// method
 	method := "getBorValidators"
 
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
What is this for?

Reply:
Normal `context.Background()` never gets cancelled automatically. It's better to cancel we are done with using it. Forgot to use it before!

Reply:
Got it